proxy/apiaxle-proxy.coffee

```diff
@@ -292,13 +294,43 @@ class exports.ApiaxleProxy extends AxleApp
     # here's the actual setting
     return endpointUrl
 
+  chooseLimits: ( key, api ) =>
+    # if keyless, use api's keyless limits
+    if key.id.substring(0, 3) == 'ip-'
+      qps = api.data.keylessQps
+      qpm = api.data.keylessQpm
+      qpd = api.data.keylessQpd
+    else
+      # use key's api specific limits if they exist
+      if key.data.apiLimits
+        try
+          keyApiLimits = JSON.parse key.data.apiLimits
+      if keyApiLimits && keyApiLimits[api.id] != undefined
+        qps = keyApiLimits[api.id].qps
+        qpm = keyApiLimits[api.id].qpm
+        qpd = keyApiLimits[api.id].qpd
+
+      # fall back to key limits
+      if qps == undefined
+        qps = key.data.qps
+      if qpm == undefined
+        qpm = key.data.qpm
+      if qpd == undefined
+        qpd = key.data.qpd
```

Comment:
it triggers if qps has not been set yet, and if key.data doesn't have 'qps', then setting qps to key.data.qps leaves it still undefined.